cmd/alpha/root.go

```diff
@@ -0,0 +1,331 @@
+		if err != nil {
+			log.Fatal(err)
+		}
+		_ = result
+	},
+}
+
+var cmdListResources = &cobra.Command{
+	Use:   "resources-list",
+	Short: "List all resources in the subscription.",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		listResources()
+	},
+}
+
+var cmdLogin = &cobra.Command{
+	Use:   "login",
+	Short: "Login to Azure using the device code login flow.",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		login()
+	},
+}
+
+func deployTemplateSubscription(deploymentName, location string, template, parameters interface{}) (res resources.DeploymentExtended, err error) {
+
+	config := NewConfig()
+	client := resources.NewDeploymentsClient(config.SubscriptionID)
+	client.Authorizer = config.Authorizer
+
+	ctx := context.Background()
+	deployment := resources.Deployment{
+		Location: &location,
+		Properties: &resources.DeploymentProperties{
+			Template:   template,
+			Parameters: parameters,
+			Mode:       resources.Incremental,
+		},
+	}
+	future, err := client.CreateOrUpdateAtSubscriptionScope(ctx, deploymentName, deployment)
+	if err != nil {
+		return res, fmt.Errorf("cannot create deployment: %v", err)
+	}
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return res, fmt.Errorf("cannot get the create deployment future response: %v", err)
+	}
+	return future.Result(client)
+
+}
+
+func deployTemplateGroup(deploymentName, resourceGroup string, template, parameters interface{}) (res resources.DeploymentExtended, err error) {
+
+	config := NewConfig()
+	client := resources.NewDeploymentsClient(config.SubscriptionID)
+	client.Authorizer = config.Authorizer
+
+	ctx := context.Background()
+	deployment := resources.Deployment{
+		Properties: &resources.DeploymentProperties{
+			Template:   template,
+			Parameters: parameters,
+			Mode:       resources.Complete,
+		},
+	}
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, deploymentName, deployment)
+	if err != nil {
+		return res, fmt.Errorf("cannot create deployment: %v", err)
+	}
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return res, fmt.Errorf("cannot get the create deployment future response: %v", err)
+	}
+	return future.Result(client)
+
+}
+
+func listResources() {
+
+	config := NewConfig()
+	client := resources.NewClient(config.SubscriptionID)
+	client.Authorizer = config.Authorizer
+
+	// list all resources
+	for result, err := client.ListComplete(context.Background(), "", "", nil); result.NotDone(); err = result.Next() {
+		if err != nil {
+			log.Fatal(err)
+		}
+		b, _ := json.Marshal(result.Value())
+
+		fmt.Printf("%s\n", b)
+	}
+
+}
+
+func login() {
+
+	config := NewConfig()
+	client := resources.NewClient(config.SubscriptionID)
+	client.Authorizer = config.Authorizer
+
+}
+
+type Config struct {
+	ClientID       string
+	TenantID       string
+	SubscriptionID string
+	Authorizer     autorest.Authorizer
+}
+
+func NewConfig() *Config {
```

Comment:
I popped this into a Config struct but I want to lean on viper more for these things going forward.